templates/nginx-stream.conf.j2

```diff
@@ -0,0 +1,18 @@
+upstream restund {
+    least_conn;
+    zone backend 64k;
+    server {{ hostvars[inventory_hostname]['ansible_' + restund_network_interface].ipv4.address }}:{{ restund_tcp_listen_port }} weight=1 max_fails=1 fail_timeout=10s;
+}
+
+# see also https://docs.nginx.com/nginx/admin-guide/security-controls/terminating-ssl-tcp/
+server {
+    listen {{ restund_tls_listen_port }} ssl;
+    proxy_pass restund;
```

Comment:
My nginx "foo" is a little outdated but have we tried this out? The docs point to nginx plus, which is a little different from the OSS version; in particular, [proxy_pass](http://nginx.org/en/docs/http/ngx_http_proxy_module.html#proxy_pass) is allowed only inside a location block and should contain a URL. If I try this config I get something like: `nginx: [emerg] "proxy_pass" directive is not allowed here`

Since this is not an HTTPS proxy, we probably need the stream solution, which requires building extra components as described [here](http://nginx.org/en/docs/stream/ngx_stream_ssl_module.html).

Maybe I missed something here 😬  I see the role installs nginx plus but we can't use it as we have no license?

Reply:
> My nginx "foo" is a little outdated but have we tried this out? The docs point to nginx plus, which is a little different from the OSS version; in particular, [proxy_pass](http://nginx.org/en/docs/http/ngx_http_proxy_module.html#proxy_pass) is allowed only inside a location block and should contain a URL. If I try this config I get something like: `nginx: [emerg] "proxy_pass" directive is not allowed here`
> 
> Since this is not an HTTPS proxy, we probably need the stream solution, which requires building extra components as described [here](http://nginx.org/en/docs/stream/ngx_stream_ssl_module.html).

See the main nginx file which includes this one inside the `stream` directive:

https://github.com/wireapp/ansible-restund/pull/11/files#diff-754bfbe9e8e9c8a0ca748ca86030662eR32-R34

The current configuration terminates TLS and forwards traffic to the TCP endpoint of restund. That part is working:

```
openssl s_client -showcerts -connect restund01.berkelium.wire.link:5349
```

Of course it would be great to have an AVS client make a connection here and check this actually works through nginx, but we don't currently have the test infrastructure here.

> Maybe I missed something here grimacing I see the role installs nginx plus but we can't use it as we have no license?

The role installs nginx the open source version. See https://github.com/nginxinc/ansible-role-nginx/blob/master/defaults/main/main.yml#L16

Reply:
OK, so it installs it with the customisations already 👍